cmake/FindGROMACS.cmake

```diff
@@ -1,22 +1,78 @@
-# - Find GROMACS
-# Find the native GROMACS includes and library
+# - Finds parts of gromacs
+# Find the native gromacs compents headers and libraries.
 #
-# GROMACS_INCLUDES - where to find .h
-# GROMACS_LIBRARIES - List of libraries when using GROMACS.
-# GROMACS_FOUND - True if GROMACS found.
+#  GROMACS_INCLUDE_DIRS - where to find gromacs headers.
+#  GROMACS_LIBRARIES    - List of libraries when used by gromacs.
+#  GROMACS_FOUND        - True if all gromacs componets were found.
+#  GROMACS_DEFINITIONS  - Extra definies needed by gromacs
+#  GROMACS_VERSION      - Gromacs lib interface version
+#
+# Copyright 2009-2018 The VOTCA Development Team (http://www.votca.org)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+function(_GROMACS_GET_VERSION _OUT_ver _version_hdr)
+  file(STRINGS ${_version_hdr} _contents REGEX "#define GMX_VERSION[ \t]+")
+  if(_contents)
+    string(REGEX REPLACE ".*#define GMX_VERSION[ \t]+([0-9.]+).*" "\\1" ${_OUT_ver} "${_contents}")
+    if(("${${_OUT_ver}}" STREUQAL "") OR (NOT ${${_OUT_ver}} MATCHES "[0-9]+"))
+        message(FATAL_ERROR "Version parsing failed for GMX_VERSION in ${_version_hdr}!")
+    endif()
+    set(${_OUT_ver} ${${_OUT_ver}} PARENT_SCOPE)
+ elseif(EXISTS ${_version_hdr})
+    message(FATAL_ERROR "No GMX_VERSION in ${_version_hdr}")
+ else()
+     message(FATAL_ERROR "No GMX_VERSION line found in include file ${_version_hdr}")
+  endif()
+endfunction()
+
+find_package(PkgConfig)
+pkg_check_modules(PC_GROMACS_D libgromacs_d)
+pkg_check_modules(PC_GROMACS libgromacs)
 
-if (GROMACS_INCLUDES)
-  # Already in cache, be silent
-  set (GROMACS_FIND_QUIETLY TRUE)
-endif (GROMACS_INCLUDES)
+find_library(GROMACS_LIBRARY NAMES gromacs_d gromacs HINTS ${PC_GROMACS_D_LIBRARY_DIRS} ${PC_GROMACS_LIBRARY_DIRS})
+if (GROMACS_LIBRARY AND NOT GROMACS_LIBRARY STREQUAL "gromacs")
+  include(CheckCXXLibraryExists)
+  check_cxx_library_exists("${GROMACS_LIBRARY}" gmx_version "" FOUND_GROMACS_VERSION_CXX)
```

Comment:
I think you need to add https://github.com/votca/csg/blob/master/CMakeModules/CheckCXXLibraryExists.cmake